api/Controllers/MiniUrl.cs

```diff
@@ -14,25 +11,20 @@ public class MiniUrlController : ControllerBase
 {
     
     private readonly ILogger<MiniUrlController> _logger;
-    private readonly IConfiguration _config;
-    private readonly string _domain;
-    private readonly IDatabase _redis0; // shortened url store
-    private readonly IDatabase _redis1; // rate limiting management store
+    private readonly string? _domain;
+    private readonly RedisActions _redis;
     
     
     public MiniUrlController(IConfiguration configuration, ILogger<MiniUrlController> logger, IConnectionMultiplexer multiplexer)
     {
-
-        _config = configuration;
         _logger = logger;
-        _domain = _config["Domain"];
-        _redis0 = multiplexer.GetDatabase(0);
-        _redis1 = multiplexer.GetDatabase(1);
+        _domain = configuration["Domain"];
+        _redis = new RedisActions(multiplexer.GetDatabase(0), multiplexer.GetDatabase(1));
```

Comment:
Since you encapsulated redis related stuff to separate class, it makes sense to also create interface for it and inject it, so your controller didn't have logic related to Redis.
Also, I would name class not RedisActions. Today it is redis DB, tomorrow it could be persisted in Postgres. 
It can be for example UrlService, UrlStore or something else :))

Reply:
done